providers/dnsimple/dnsimpleProvider.go

```diff
@@ -199,6 +209,34 @@ func (c *DnsimpleApi) GetRegistrarCorrections(dc *models.DomainConfig) ([]*model
 	return corrections, nil
 }
 
+// getDNSSECCorrections returns corrections that update a domain's DNSSEC state.
+func (c *DnsimpleApi) getDNSSECCorrections(dc *models.DomainConfig) ([]*models.Correction, error) {
+	enabled, err := c.getDnssec(dc.Name)
+	if err != nil {
+		return nil, err
+	}
+
+	if enabled && !dc.AutoDNSSEC {
+		return []*models.Correction{
+			{
+				Msg: "Disable DNSSEC",
+				F:   func() error { _, err := c.disableDnssec(dc.Name); return err },
```

Comment:
Will dc.Name be the right contents by the time this function is called?  I think so, but it is safer to have `domain := dc.Name` early in the parent function, then refer to domain in the func()'s. 

I can be talked out of this. I'm not an expert in this area. I just know I've hit this in the past with other things.

Reply:
Yes it's fine: you're thinking of loops which mutate the value being bound because people aren't capturing the value for the loop.  In this case, there's no looping and if someone changes the domain name underneath us ... all bets are off.  In this case, I'm matching existing style of several matching functions in this file.

So I'd prefer to match existing style, if that's acceptable?

Reply:
Yes, leave it as is.